### PR TITLE
feat: Added pool breakdown of operator vs member stake, added option to add…

### PR DESCRIPTION
**Pull request by alexkroeger**
… fees to epoch

Changes:
- Added an option to add fees to the epoch endpoint. Fees make the amount of data queried much larger, but they are useful for estimating rewards for an epoch.
- Added a breakdown of stake for the /pools endpoint between members and operator.

Testing:
- [x] Tested locally on Hashalytics

Dependencies:
https://github.com/0xProject/0x-event-pipeline/pull/24

### CODE REVIEW

File: src/handlers/staking_handlers.ts [hunk 1]
```diff
@@ -39,14 +40,37 @@ export class StakingHandlers {
...
-        const response: StakingEpochsResponse = {
-            currentEpoch,
-            nextEpoch,
-        };
+        // optional query string to include fees
+        let isWithFees: boolean;
+        if (_req.query.withFees) {
+            if (!(_req.query.withFees === 'true' || _req.query.withFees === 'false')) {
+                res.status(HttpStatus.BAD_REQUEST).send(`Invalid value for withFees`);
+            }
```
Comment (fragosti) [inline src/handlers/staking_handlers.ts:48]: The reason `_req` had the underscore is that it wasn't being used. Now that it is being used, we need to remove the underscore.

File: src/handlers/staking_handlers.ts [hunk 2]
```diff
@@ -39,14 +40,37 @@ export class StakingHandlers {
...
+            isWithFees = _req.query.withFees === 'true';
+        } else {
+            isWithFees = false;
+        }
```
Comment (fragosti) [inline src/handlers/staking_handlers.ts:52]: Elsewhere we extract query params in pure functions (so we can actually type the query object)

File: src/services/staking_data_service.ts [hunk 3]
```diff
@@ -41,15 +43,35 @@ export class StakingDataService {
...
+        );
+        if (!rawEpochWithFees) {
+            throw new Error('Could not find the current epoch.');
+        }
+        const epoch = stakingUtils.getEpochWithFeesFromRaw(rawEpochWithFees);
+        return epoch;
+    }
+
     public async getNextEpochAsync(): Promise<Epoch> {
         const rawEpoch: RawEpoch | undefined = _.head(await this._connection.query(queries.nextEpochQuery));
```
Comment (fragosti) [inline src/services/staking_data_service.ts:58]: Are we now over-fetching information for this query?

File: src/handlers/staking_handlers.ts (hunk 2 above)
Comment (alexkroeger) [reply src/handlers/staking_handlers.ts:52]: 👍

File: src/handlers/staking_handlers.ts (hunk 1 above)
Comment (alexkroeger) [reply src/handlers/staking_handlers.ts:48]: 👍

File: src/services/staking_data_service.ts (hunk 3 above)
Comment (alexkroeger) [reply src/services/staking_data_service.ts:58]: nah, it runs a separate query that doesn't SUM over fills

**Comment by alexkroeger** [comment]
deploy staging

**Pull request merged**